lib/Dialect/Torch/IR/TorchOps.cpp

```diff
@@ -819,14 +819,33 @@ void AtenAddTensorOp::getCanonicalizationPatterns(RewritePatternSet &patterns,
 // AtenSizeOp
 //===----------------------------------------------------------------------===//
 
+static FailureOr<BaseTensorType> traceKnownSizeTensorType(Value value) {
+  Type type = value.getType();
+  if (!type.isa<BaseTensorType>())
+    return failure();
+
+  auto baseTensorType = type.cast<BaseTensorType>();
+  if (baseTensorType.areAllSizesKnown())
+    return baseTensorType;
+
+  auto op = value.getDefiningOp();
```

Comment:
please write this as a while loop, and bail out after "6" (yes, hardcoded ok) iterations. We cannot have an unbounded IR traversal since it can create superlinear compilation time.

Reply:
Ah right, I had not thought of the effect on compilation times due to deep hierarchies.  Fixed.